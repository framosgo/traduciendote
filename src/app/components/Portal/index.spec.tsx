import * as React from 'react';
import { shallow } from 'enzyme';

import Portal from '.';

beforeEach(() => {
  // @ts-ignore
  const modalRoot = global.document.createElement('div');
  modalRoot.setAttribute('id', 'modal-root');

  // @ts-ignore
  const body = global.document.querySelector('body');
  body.appendChild(modalRoot);
});

afterEach(() => {
  // @ts-ignore
  const modalRoot = global.document.getElementById('modal-root');
  if (modalRoot) {
    modalRoot.remove();
  }
});

describe('Portal', () => {
  it('should return a valid portal', () => {
    const component = shallow(<Portal>Content</Portal>);

    expect(component).toMatchSnapshot();
  });

  it('should return a valid portal with two childs', () => {
    const component = shallow(
      <div>
        <Portal>Content</Portal>
        <Portal>Content 2</Portal>
      </div>
    );

    expect(component).toMatchSnapshot();
  });
});
