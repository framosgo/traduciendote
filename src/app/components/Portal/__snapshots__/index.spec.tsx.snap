// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal should return a valid portal 1`] = `
<Portal
  containerInfo={<div />}
>
  Content
</Portal>
`;

exports[`Portal should return a valid portal with two childs 1`] = `
<div>
  <Portal>
    Content
  </Portal>
  <Portal>
    Content 2
  </Portal>
</div>
`;
